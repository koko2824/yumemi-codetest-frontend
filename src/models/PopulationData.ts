export interface PopulationData {
  year: number;
  [prefecture: string]: number;
}
