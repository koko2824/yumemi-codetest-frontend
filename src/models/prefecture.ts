export interface Prefecture {
  prefCode: number;
  prefName: string;
}

export interface PrefecturesResponse {
  message: string | null;
  result: Prefecture[];
}
