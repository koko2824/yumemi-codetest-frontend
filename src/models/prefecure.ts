export interface Prefecure {
  prefCode: number;
  prefName: string;
}
