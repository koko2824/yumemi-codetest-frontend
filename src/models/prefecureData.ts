export interface PrefecureData {
  prefCode: number;
  preName: string;
}
