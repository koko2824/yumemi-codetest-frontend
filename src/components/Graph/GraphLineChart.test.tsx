import React from 'react';
import { render } from '@testing-library/react';
import PopulationLineChart from './GraphLineChart';

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}

global.ResizeObserver = ResizeObserver;
import { ChartData } from '@/models/PopulationData';
import { Prefecture } from '@/models/prefecture';

const mockChartData: ChartData[] = [
  { year: 2000, population: 1000 },
  { year: 2005, population: 1500 },
  { year: 2010, population: 2000 },
];

const mockSelectedPrefectures: Prefecture[] = [
  { prefCode: 1, prefName: '北海道' },
  { prefCode: 13, prefName: '東京都' },
  { prefCode: 47, prefName: '沖縄県' },
];

export const mockGetLineColor = (index: number) => {
  const colors = [
    '#FF6384',
    '#36A2EB',
    '#FFCE56',
    '#4BC0C0',
    '#9966FF',
    '#FF9F40',
    '#8AC926',
    '#1982C4',
    '#6A4C93',
    '#F94144',
  ];
  return colors[index % colors.length];
};

describe('PopulationLineChart コンポーネント', () => {
  it('正しくレンダリングされているか', () => {
    const { container } = render(
        <PopulationLineChart
            chartData={mockChartData}
            selectedPrefectures={mockSelectedPrefectures}
            getLineColor={mockGetLineColor}
        />
    );
    expect(container).toBeInTheDocument();
  });

  it('ラインカラーが正しく設定されているか', () => {
    const { container } = render(
      <PopulationLineChart
        chartData={mockChartData}
        selectedPrefectures={mockSelectedPrefectures}
        getLineColor={mockGetLineColor}
      />
    );
    const lines = container.querySelectorAll('.recharts-line path');
    lines.forEach((line, index) => {
      expect(line).toHaveAttribute('stroke', mockGetLineColor(index));
    });
  });

  it('ウィンドウサイズが変更された場合に正しく再レンダリングされるか', () => {
    const { container } = render(
      <PopulationLineChart
        chartData={mockChartData}
        selectedPrefectures={mockSelectedPrefectures}
        getLineColor={mockGetLineColor}
      />
    );

    global.dispatchEvent(new Event('resize'));
    expect(container).toBeInTheDocument();
  });
});
