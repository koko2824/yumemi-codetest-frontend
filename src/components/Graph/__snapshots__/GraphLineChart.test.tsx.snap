// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopulationLineChart コンポーネント 正しくレンダリングされているか 1`] = `
<div>
  <div
    style="width: 500px; height: 400px;"
  >
    <div
      class="h-[400px] w-full"
      style="min-width: 300px; min-height: 400px;"
    >
      <div
        class="recharts-responsive-container"
        style="width: 100%; height: 100%; min-width: 0;"
      />
    </div>
  </div>
</div>
`;
